libs/wingc/src/jsify.rs

```diff
@@ -710,6 +695,315 @@ impl JSifier {
+			self.jsify_symbol(name),
+			if let Some(parent) = parent {
+				format!(" extends {}", self.jsify_type(parent))
+			} else {
+				format!(" extends {}", WINGSDK_RESOURCE)
+			},
+			self.jsify_resource_constructor(constructor, parent.is_none()),
+			preflight_methods
+				.iter()
+				.map(|(n, m)| format!(
+					"{} = {}",
+					n.name,
+					self.jsify_function(None, &m.parameters, &m.statements, phase)
+				))
+				.collect::<Vec<String>>()
+				.join("\n"),
+			toinflight_method
+		);
+
+		// For each inflight methods generate an annotation which includes a list of all the captured
+		// fields and all the ops they provide.
+		// TODO: in the future we should only pass the relevant captured fields and ops to each method.
+		let default_annotation = captured_fields
+			.iter()
+			.map(|(name, _, ops)| {
+				format!(
+					"\"this.{}\": {{ ops: [{}]}}",
+					name,
+					ops.iter().map(|op| format!("\"{}\"", op)).collect_vec().join(", ")
+				)
+			})
+			.collect_vec()
+			.join(", ");
+		let inflight_annotations = inflight_methods
+			.iter()
+			.map(|(method_name, ..)| {
+				format!(
+					"{}._annotateInflight(\"{}\", {{{}}});",
+					name.name, method_name.name, default_annotation
+				)
+			})
+			.collect_vec();
+
+		// Return the preflight resource class
+		return format!("{}\n{}", resource_class, inflight_annotations.join("\n"));
+	}
+
+	fn jsify_resource_constructor(&self, constructor: &Constructor, no_parent: bool) -> String {
+		let mut lines = vec![];
+		lines.push("{".to_string());
+
+		// If there's no parent then this resource id derived from the base resource class (core.Resource) and we need
+		// to manually call its super
+		if no_parent {
+			lines.push("	super(scope, id);".to_string());
+		}
+
+		for statement in constructor.statements.statements.iter() {
+			let statement_str = self.jsify_statement(
+				constructor.statements.env.borrow().as_ref().unwrap(),
+				statement,
+				Phase::Preflight,
+			);
+			let result = statement_str.split("\n");
+			for l in result {
+				lines.push(format!("  {}", l));
+			}
+		}
+
+		lines.push("}".to_string());
+
+		format!(
+			"constructor(scope, id, {}) {{\n{}\n}}",
+			constructor
+				.parameters
+				.iter()
+				.map(|(name, _)| name.name.clone())
+				.collect::<Vec<_>>()
+				.join(", "),
+			lines.join("\n")
+		)
+	}
+
+	fn jsify_toinflight_method(
+		&self,
+		resource_name: &Symbol,
+		captured_fields: &[(String, TypeRef, Vec<String>)],
+	) -> String {
+		let inner_clients = captured_fields
+			.iter()
+			.map(|(inner_member_name, _, _)| {
+				format!(
+					"const {}_client = this.{}._toInflight();",
+					inner_member_name, inner_member_name,
+				)
+			})
+			.collect::<Vec<_>>();
+
+		let client_relative_path = format!("clients/{}.inflight.js", resource_name.name);
+		format!(
+			"_toInflight() {{\n{}\nconst self_client_path = {};\nreturn core.NodeJsCode.fromInline(`(new (require(\"${{self_client_path}}\")).{}_inflight({{{}}}))`);\n}}",
```

Comment:
Can we perhaps make this more readable by using an array of strings and joining them using `\n`?

Reply:
ended up using [indoc](https://github.com/dtolnay/indoc)
should eventually change a lot of the jsification to use it, pretty cool.